### ISSUE TITLE
sddm: fails to start with xauth error

### ISSUE THREAD

**Issue by dtulig** [NixOS/nixpkgs#10211]
SDDM will fail to start when built with a recent unstable channel (https://github.com/NixOS/nixpkgs-channels/commit/e916273209560b302ab231606babf5ce1c481f08).

X starts up and outputs this error message:

```
systemd[1]: Starting X11 Server...
systemd[1]: Started X11 Server.
sddm[582]: Initializing...
sddm[582]: Starting...
sddm[582]: Adding new display 0 on vt 1 ...
sddm[582]: Adding cookie to "/var/run/sddm/:0"
display-manager-start[582]: sh: /usr/bin/xauth: No such file or directory
sddm[582]: Display server starting...
sddm[582]: Running: /usr/bin/X :0 -auth /var/run/sddm/:0 -nolisten tcp -background none -noreset vt1
sddm[582]: Failed to start display server process
```

Reverting https://github.com/NixOS/nixpkgs-channels/commit/60fc0af74ad9e4c241f27ef41e5e1a4e45c57e14 resolves the issue and allows SDDM to start again.

My leading theory is that sddm is ignoring `/etc/sddm.conf`. That file is present and has the proper paths and settings, including for xauth.

**Comment by Mathnerd314** [comment]
ping @ttuegel since I don't think referencing the commit is enough

**Comment by ttuegel** [comment]
> My leading theory is that sddm is ignoring /etc/sddm.conf.

Yes, that seems to be the problem. I'll see if I can figure out why...

**Comment by ttuegel** [comment]
This is fixed *properly* in 6a80bb9. The config file was not loaded because it was "too old." It wasn't actually, but its mtime was being compared against an uninitialized value.

**Comment by Mathnerd314** [comment]
After looking at this some more:
* Per https://bugreports.qt.io/browse/QTBUG-24831, the reported file properties are always the target of the symlink, even if it's Windows and the documentation explicitly states that links are not dereferenced. So in our case we get msecs=1 in the Nix store. In contrast, the symlink has a useful timestamp, but this is not accessible at all through the Qt API.
* QDateTime is initialized to a "null" date/time, which is 0 msecs after the epoch, but in local time. This then gets converted to UTC, giving msecs=60*60*1000*7, which is larger than 1, so the file is not loaded.

So, removing the check is the only reasonable option, modulo hacking Qt.
